src/base/bittorrent/peerinfo.cpp

```diff
@@ -408,3 +408,8 @@ QString PeerInfo::flagsDescription() const
 {
     return m_flagsDescription;
 }
+
+int PeerInfo::downloadingPieceIndex() const
```

Comment:
Thank you for the answers! 

This has to do with the second use case. The use case of the progress bar. There I show piece contents and file extension over pieces in the torrent progress bars. There are no peers in that case. There might be no downloading process either. I need just static information about the torrent layout. Needed functions come from the `torrent_info` class and I added an utility function to the `TorrentInfo` class.

From the answer  you gave me I see now that this is in agreement with your view too.